infra.bs

```diff
@@ -557,6 +557,49 @@ the end result is indistinguishable. For instance, as long as |result| in the ab
 mutated during a compute operation, the user agent could stop the computation.
 
 
+<h3 id=algorithm-conditional-statements>Conditional statements</h3>
+
+<p>Algorithms with conditional statements should use the keywords "if", "then", and "otherwise".
+
+<div class=example id=example-conditional-statement>
+ <ol>
+  <li><p>Let |value| be null.
+
+  <li><p>If |input| is a <a>string</a>, then set |value| to |input|.
+
+  <li><p>Return |value|.
+ </ol>
+</div>
+
+Once the keyword "otherwise" is used, the keyword "then" is omitted.
+
+<div class=example id=example-conditional-statement-no-then-1>
+ <ol>
+  <li><p>Let |value| be null.
+
+  <li><p>If |input| is a <a>string</a>, then set |value| to |input|.
+
+  <li><p>Otherwise, set |value| to failure.
+
+  <li><p>Return |value|.
```

Comment:
You could add a step before this that asserts that value is not null. It's a bit obvious here, but the suggestion might help people writing more complicated algorithms.